### DISCUSSION TITLE
Is there a significant performance difference between using a ClusterIP service and a headless service  when only using istio as gateway (without enabling serive-mesh)?

### DISCUSSION THREAD

**Discussion by caozhuozi** [istio/istio#53851] [Q&A] [answered]
as titled

**Comment by howardjohn** [comment, marked as answer]
There should be absolutely no difference in the datapath when connecting to a headless or clusterIP service through an istio gateway

**Reply by caozhuozi** [reply]
Thanks @howardjohn for keeping an eye on these small questions! 😊

Could you explain more why no performance difference here? I think istio gateway has to do an addtional DNS resolustion for headless service?

**Reply by howardjohn** [reply]
Istio discovery watchers the API server for endpoints. It does not rely on DNS. So these cases are basically the same
